docs/release_process.md

```diff
@@ -0,0 +1,80 @@
+# Release Process
+
+So you want to release a new version, great!
+
+First, make sure all the desired pull requests have been merged.
+
+Next, pull [the latest translations](translate.md).
```

Comment:
> We don't have to just pull the latest translations. We have to push them to transiflex, be sure they are translated and pull them.

I don't think the release is the right moment to push the translations and wait for translators to do their magic.

We should do that much earlier.

When comes the time to make a release, we can only pull what has been translated up to now.